hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalMigration.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.mirror.importer.migration;
+
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.importer.MirrorProperties;
+import com.hedera.mirror.importer.db.DBProperties;
+import com.hederahashgraph.api.proto.java.Key;
+import jakarta.inject.Named;
+import java.sql.ResultSet;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+import lombok.NoArgsConstructor;
+import lombok.SneakyThrows;
+import org.flywaydb.core.api.MigrationVersion;
+import org.springframework.context.annotation.Lazy;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.transaction.support.TransactionOperations;
+
+@Named
+public class SyntheticCryptoTransferApprovalMigration extends AsyncJavaMigration<Long> {
+
+    private static final Long LOWER_BOUND_TIMESTAMP = 1680284879342064922L;
+    private static final Long UPPER_BOUND_TIMESTAMP = 1686243920981874003L;
```

Comment:
```suggestion
    private static final Long UPPER_BOUND_TIMESTAMP = 1686243920981874002L;
```

make upper bound the timestamp of first transaction with HAPI 0.38.x minus 1, then in the SQL change it to ` <= :upper_bound`

otherwise `migrationPartial` will always miss the transaction at the upper bound consensus timestamp for each iteration

Reply:
Thanks I have updated the number to be 1 less than the timestamp.